src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseErrorPageMiddleware.cs

```diff
@@ -131,6 +132,12 @@ public virtual async Task Invoke(HttpContext httpContext)
                             {
                                 var databaseExists = await relationalDatabaseCreator.ExistsAsync();
 
+                                if (databaseExists)
+                                {
+                                    // Also check if the database is completely empty - see https://github.com/aspnet/EntityFrameworkCore/issues/15997
+                                    databaseExists = (bool)typeof(RelationalDatabaseCreator).GetMethod("HasTables", BindingFlags.Instance | BindingFlags.NonPublic).Invoke(relationalDatabaseCreator, null);
```

Comment:
Cache the `MethodInfo`

Reply:
This is only temporary until I get the fix for https://github.com/aspnet/EntityFrameworkCore/issues/15997 checked in.

Reply:
Ok and you can remove the cached value when that is fixed.